### ISSUE TITLE
Consider using EXISTS for count query

### ISSUE THREAD

**Issue by punktilious** [IBM/FHIR#1244]
**Describe the bug**
```
SELECT COUNT(R.RESOURCE_ID)  FROM Basic_LOGICAL_RESOURCES LR 
JOIN Basic_RESOURCES R 
   ON R.LOGICAL_RESOURCE_ID=LR.LOGICAL_RESOURCE_ID 
 AND R.RESOURCE_ID = LR.CURRENT_RESOURCE_ID AND R.IS_DELETED <> 'Y'  
JOIN (SELECT DISTINCT LOGICAL_RESOURCE_ID FROM Basic_TOKEN_VALUES  
           WHERE (PARAMETER_NAME_ID=39 AND ((TOKEN_VALUE = ?)))) Basic0 
ON Basic0.LOGICAL_RESOURCE_ID = R.LOGICAL_RESOURCE_ID
```

The DISTINCT in the sub-query might prevent the optimizer from picking a better plan. Consider using `EXISTS` and correlate the sub-query (without distinct) with the main body. Semantically the result should be the same.

**To Reproduce**
Run JDBCPagingTest as an example.

**Expected behavior**
Best possible performance.
**Additional context**

**Comment by punktilious** [comment]
This has been implemented as part of #1385

**Comment by prb112** [comment]
Verified during Code Review, the queries do not generate the DISTINCT and use the EXISTS approach to counting.  As discussed on the Team call, I am closing this issue.
